packages/transformers/js/core/src/modules.rs

```diff
@@ -594,7 +595,7 @@ impl Fold for ESMFold {
     if self.in_export_decl {
       self.create_export(
         node.key.sym.clone(),
-        Expr::Ident(node.key.clone()),
+        Expr::Ident(node.key.id.clone()),
```

Comment:
Just sharing what I shared in Discord here, it seems that the change arround `AssignPatProp` means that after visiting here it will visit into `fold_binding_intent`, this results in a duplicate export being added for exports of the form `export const { Foo } = bar;` in non-scope hoisting mode - this breaks at runtime.

This change fixed the issue for me, but not sure if there's some edge case that it might not be correct for.  It would be good to start with a failing integration test.

```
diff --git a/packages/transformers/js/core/src/modules.rs b/packages/transformers/js/core/src/modules.rs
index 65235f3c3..bd13d10b6 100644
--- a/packages/transformers/js/core/src/modules.rs
+++ b/packages/transformers/js/core/src/modules.rs
@@ -598,9 +598,11 @@ impl Fold for ESMFold {
         Expr::Ident(node.key.id.clone()),
         DUMMY_SP,
       );
+      node
+    } else {
+      node.fold_children_with(self)
     }
 
-    node.fold_children_with(self)
   }
 
   modules_visit_fn!(fold_function, Function);
```